library/x509_crt.c

```diff
@@ -1556,7 +1557,16 @@ int mbedtls_x509_crt_parse_path(mbedtls_x509_crt *chain, const char *path)
     p = filename + len;
     filename[len++] = '*';
 
-    w_ret = MultiByteToWideChar(CP_ACP, 0, filename, (int) len, szDir,
+    if (FAILED(SizeTToInt(len, &length_as_int))) {
+        return MBEDTLS_ERR_X509_FILE_IO_ERROR;
+    }
+
+    /*
+     * Note this function uses the code page CP_ACP which is the system default
+     * ANSI codepage. The input string is always described in BYTES and the
+     * output length is described in WCHARs.
+     */
+    w_ret = MultiByteToWideChar(CP_ACP, 0, filename, length_as_int, szDir,
```

Comment:
Re-reading this change, I'm not sure why we're bothering with the whole `length_as_int` thing.

This would be a concern if `len` wouldn't fit in an `int`, but we actually know that `len` <= MAX_PATH - 3`, which is 260 (https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry)

So the cast in the (original) line 1559 is totally fine.

We also do a similar cast in (new) line 1588 - admittedly this is now "length remaining in buffer", which is a horrible re-use of variable with opposite meaning, but if we are happy with the cast at 1588 we should be happy with the cast at 1559.

(And fewer changes => better, IMO)